### PR TITLE
Add ipydatawidgets recipe

### PR DESCRIPTION
**Pull request by vidartf**

### CODE REVIEW

File: recipes/ipydatawidgets/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{% set name = "ipydatawidgets" %}
+{% set version = "3.1.0" %}
+{% set sha256 = "973370568799469ed2a6c402f93e2fa8a63c74e044ea1959def3261fb6158464" %}
+# sha256 is the prefered checksum -- you can get it for a file with:
```
Comment (mariusvniekerk) [inline recipes/ipydatawidgets/meta.yaml]: remove these comments please

Comment (vidartf) [reply recipes/ipydatawidgets/meta.yaml]: Done.

**Comment by vidartf** [comment]
After notebook 5.3, linking steps are not needed (notebook.d config file as a data file). This does mean that any user with notebook < 5.3 will need to enable the classic notebook extension manually if they want to use it. That being said, I'd rather prefer that over the classic link scripts, as those will cause an issue when transitioning away from them in the future (the unlink step of the old version will prevent the new config system from working as intended).

**Comment by mariusvniekerk** [comment]
Cool that'll work then

File: recipes/ipydatawidgets/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+requirements:
+  host:
+    - python
+    - pip
+  run:
+    - python
+    - ipywidgets >=7,<8
+    - numpy
+    - traittypes >= 0.2
```
Comment (jakirkham) [inline recipes/ipydatawidgets/meta.yaml]: This should not have a space after the `=`s, which is blocking its conversion.

Could you please send a PR to fix this, @vidartf?

Comment (vidartf) [reply recipes/ipydatawidgets/meta.yaml]: Submitted as #6031.

Comment (vidartf) [reply recipes/ipydatawidgets/meta.yaml]: Would it be reasonable to have the linter/CI look for these kind of issues?

Comment (jakirkham) [reply recipes/ipydatawidgets/meta.yaml]: Thanks for doing that.

Yeah. Honestly am surprised `conda-build` didn't reject this. Feel free to raise issues to both `conda-smithy` and `conda-build`.

**Pull request merged**